src/components/TradeWidget/index.tsx

```diff
@@ -74,17 +86,79 @@ const TradeWidget: React.FC = () => {
     }
   }
 
+  async function placeOrder(
+    buyAmount: BN,
+    buyToken: TokenDetails,
+    sellAmount: BN,
+    sellToken: TokenDetails,
+  ): Promise<void> {
+    setSubmitting(true)
+    log(
+      `Placing order: buy ${buyAmount.toString()} ${buyToken.symbol} sell ${sellAmount.toString()} ${sellToken.symbol}`,
+    )
+
+    try {
+      const sellTokenId = await exchangeApi.getTokenIdByAddress(sellToken.address)
+      const buyTokenId = await exchangeApi.getTokenIdByAddress(buyToken.address)
+      // TODO: get this value from a config, no magic numbers
+      const validUntil = (await exchangeApi.getCurrentBatchId()) + 6 // every batch takes 5min, we want it to be valid for 30min, ∴ 30/5 == 6
+
+      const params: PlaceOrderParams = {
+        userAddress,
+        buyTokenId,
+        sellTokenId,
+        validUntil,
+        buyAmount,
+        sellAmount,
+      }
+      const result = await exchangeApi.placeOrder(params, txOptionalParams)
+      log(`The transaction has been mined: ${result.receipt.transactionHash}`)
+
+      toast.success(`Placed order id=${result.data} valid for 30min`)
+
+      reset() // the form
+    } catch (e) {
+      log(`Error placing order`, e)
+      toast.error(`Error placing order: ${e.message}`)
+    } finally {
+      //TODO: use mounted hook thingy when available
+      setSubmitting(false)
+    }
+  }
+
+  async function onSubmit(data: FieldValues): Promise<void> {
+    if (isConnected) {
+      placeOrder(
+        parseAmount(data[receiveInputId], receiveToken.decimals),
+        receiveToken,
+        parseAmount(data[sellInputId], sellToken.decimals),
+        sellToken,
+      )
+    } else {
+      history.push('/connect-wallet')
```

Comment:
One way to hand over and restore the form state would be `history.push('/connect-wallet', state)`

Reply:
To avoid re-work, I'll wait for David's changes to be merged.

Reply:
Addressed in this PR https://github.com/gnosis/dex-react/pull/184